app/assets/javascripts/components/PipelineSampleReport.jsx

```diff
@@ -70,94 +150,101 @@ class PipelineSampleReport extends React.Component {
     });
   }
 
-  refreshPage(overrides) {
-    new_params = Object.assign({}, this.report_page_params, overrides);
-    window.location = location.protocol + '//' + location.host + location.pathname + '?' + jQuery.param(new_params);
-  }
-
-  disableFilters() {
-    disable_filters = 1;
-    ReportFilter.showLoading('Disabling filters...');
-    this.refreshPage({disable_filters});
-  }
-
-  enableFilters() {
-    disable_filters = 0;
-    ReportFilter.showLoading('Enabling filters...');
-    this.refreshPage({disable_filters});
-  }
-
   // applySort needs to be bound at time of use, not in constructor above
+  // TODO(yf): fix this
   applySort(sort_by) {
     const regex = new RegExp('_', 'g');
     ReportFilter.showLoading(`Sorting the table by ${sort_by.replace(regex, ' ')}...`);
     this.refreshPage({sort_by});
   }
 
-  setFilterThreshold(threshold_name, event) {
-    this.new_filter_thresholds[threshold_name] = event.target.value.trim();
-    $('.apply-filter-button a').addClass('changed');
+  setFilterThreshold(e) {
+    threshold_name = e.target.id
+    val = parseFloat(e.target.value.trim());
+    if (isNaN(val)) {
+      delete this.state.new_filter_thresholds[threshold_name]
+    } else {
+      this.state.new_filter_thresholds[threshold_name] = val;
```

Comment:
I'm assuming it's intentional but worth noting that modifying the state directly like this won't trigger a re-render